modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpIgniteClient.java

```diff
@@ -235,13 +235,17 @@ private TcpIgniteClient(ClientConfiguration cfg) throws ClientException {
         ensureCacheName(name);
 
         ch.request(ClientOperation.CACHE_DESTROY, req -> req.out().writeInt(ClientUtils.cacheId(name)));
+        ch.unregisterKeyPartitionMapperFactory(ClientUtils.cacheId(name));
```

Comment:
This will only work if cache create and destroy were made on the same client. What if cache was destoyed on server for example?

Reply:
Fixed.